nifi-frontend/src/main/frontend/apps/nifi/src/app/ui/common/controller-service/move-controller-service/move-controller-service.component.ts

```diff
@@ -0,0 +1,158 @@
+import { CloseOnEscapeDialog } from '../../close-on-escape-dialog/close-on-escape-dialog.component';
+import { moveControllerService } from './../../../../pages/flow-designer/state/controller-services/controller-services.actions';
+import { SelectOption } from './../../../../state/shared/index';
+
+@Component({
+    selector: 'move-controller-service',
+    standalone: true,
+    templateUrl: './move-controller-service.component.html',
+    imports: [
+        ReactiveFormsModule,
+        MatDialogModule,
+        MatInputModule,
+        MatCheckboxModule,
+        MatButtonModule,
+        MatTabsModule,
+        MatOptionModule,
+        MatSelectModule,
+        ControllerServiceApi,
+        ControllerServiceReferences,
+        AsyncPipe,
+        NifiSpinnerDirective,
+        NifiTooltipDirective,
+        NgTemplateOutlet
+    ],
+    styleUrls: ['./move-controller-service.component.scss']
+})
+export class MoveControllerService extends CloseOnEscapeDialog {
+    @Input() goToReferencingComponent!: (component: ControllerServiceReferencingComponent) => void;
+    protected readonly TextTip = TextTip;
+    protected controllerServiceActionProcessGroups: SelectOption[] = [];
+
+    controllerService$ = this.store.select(selectControllerService);
+
+    moveControllerServiceForm: FormGroup;
+
+    @ViewChild('stepComplete') stepComplete!: TemplateRef<any>;
+    @ViewChild('stepError') stepError!: TemplateRef<any>;
+    @ViewChild('stepInProgress') stepInProgress!: TemplateRef<any>;
+    @ViewChild('stepNotStarted') stepNotStarted!: TemplateRef<any>;
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public request: MoveControllerServiceDialogRequest,
+        private store: Store<ControllerServiceState>,
+        private formBuilder: FormBuilder
+    ) {
+        super();
+
+        request.processGroupFlow.subscribe((flow:any) => {
+            let processGroups: SelectOption[] = [];
+            let processGroupflow = flow.processGroupFlow;
+            if (processGroupflow.breadcrumb.hasOwnProperty('parentBreadcrumb')) {
+                let parentBreadcrumb = processGroupflow.breadcrumb.parentBreadcrumb;
+                if (parentBreadcrumb.permissions.canRead && parentBreadcrumb.permissions.canWrite) {
+                    let option: SelectOption = {
+                        text: parentBreadcrumb.breadcrumb.name + ' (Parent)',
+                        value: parentBreadcrumb.breadcrumb.id
+                    }
+                    processGroups.push(option);
+                }
+            }
+
+            processGroupflow.flow.processGroups.forEach((child:any) => {
+                if (child.permissions.canRead && child.permissions.canWrite) {
+                    let option: SelectOption = {
+                        text: child.component.name,
+                        value: child.component.id
+                    }
+                    processGroups.push(option);
+                }
+            });
```

Comment:
We should only allow the user to choose a child component if the reduction in Controller Service scope will be allowed. By moving to a child Group if there are any components referencing that Controller Service in the current Group won't be able to.

Reply:
The reason I had not limited the process group list is because if the user does try and move a controller service that has referencing components that outside of the new scope, the API returns an error listing the components that have a conflict. If I filter out the list, the user won't know why it's not in the drop down.

I can change this to only display viable process groups if that is what is wanted though.

Reply:
Thoughts on adding it to the listing but disabling it and adding a description as to why it cannot be selected? The `SelectOption` already supports `disabled` and `description` field so it should just be a matter of conditionally populating the fields. Though you may need to also account for them in your template.

Reply:
I've implemented this change. The dropdown menu will still list all parent and child process groups, but any process group with a conflict will not be disabled with an ! icon and a hover tooltip explaining the conflict. 

I will note this does require the use of 2 api calls. I also had to add an optional parameter to an api definition.

Reply:
Actually, I thought of something about 2 seconds after I pushed. Found a way to avoid needed an api call. So now we only have 1 api call. Tested and pushed.

Reply:
Shouldn't we be able to determine this without any additional API calls? Within the Controller Service we have a list of referencing components which contain their Parent Process Group ID. We also have the breadcrumbs which contain the current Process Group and all of the ancestors Process Groups.

If the referencing component's Parent Group ID is one of the breadcrumbs than it would mean that the Controller Service would be out of scope if referencing component were moved to a child Process Group.